### PR TITLE
feat(eslint-config-fluid): Add strict-biome config for use in conjunction with Biome linter

### PR DESCRIPTION
**Pull request by tylerbutler**
Adds a strict config to the shared eslint package that can be used in projects to use Biome for linting in conjunction with eslint. This config is the same as the strict config but disables rules that have a Biome equivalent.

In addition, the fluid-eslint-plugin dependency is updated to its latest patch release, 0.1.3.

### CODE REVIEW

File: common/build/eslint-config-fluid/strict.js
Comment (tylerbutler) [inline common/build/eslint-config-fluid/strict.js:23]: Do not merge without reverting this.

**Review by jason-ha — COMMENTED** [review]
Couldn't we just disable formatting things generally and leave it up to formatters directly?

File: common/build/eslint-config-fluid/strict.js
Comment (tylerbutler) [reply common/build/eslint-config-fluid/strict.js:23]: I made this change so it was easier to see the diff between the strict and strict-biome configs via the printed config. Once reviewed I'll revert.

**Comment by tylerbutler** [comment]
> Couldn't we just disable formatting things generally and leave it up to formatters directly?

This change is not about formatting. It's about linting. Eslint 8 is deprecated. Eslint 9 will require work to adopt. I want to explore not using Eslint altogether, or supplementing it with a faster linter (biome, oxlint) to better inform our lint plans for the future.

**Comment by jason-ha** [comment]
> > Couldn't we just disable formatting things generally and leave it up to formatters directly?
> 
> This change is not about formatting. It's about linting. Eslint 8 is deprecated. Eslint 9 will require work to adopt. I want to explore not using Eslint altogether, or supplementing it with a faster linter (biome, oxlint) to better inform our lint plans for the future.

By "formatting things" I mean all lint formatting rules. Suggesting that we don't have a special Biome config and instead disable all of these format related rules everywhere.

**Comment by tylerbutler** [comment]
> By "formatting things" I mean all lint formatting rules. Suggesting that we don't have a special Biome config and instead disable all of these format related rules everywhere.

Maybe I am misunderstanding. The special Biome config is not disabling formatting rules exclusively. It's disabling lint rules from [this list](https://biomejs.dev/linter/rules-sources/#rules-from-other-sources) (partially - only the "recommended" rules), which includes stuff like [no explicit-any](https://biomejs.dev/linter/rules/no-explicit-any/) and [use-for-of](https://biomejs.dev/linter/rules/use-for-of/).

**Comment by jason-ha** [comment]
> > By "formatting things" I mean all lint formatting rules. Suggesting that we don't have a special Biome config and instead disable all of these format related rules everywhere.
> 
> Maybe I am misunderstanding. The special Biome config is not disabling formatting rules exclusively. It's disabling lint rules from [this list](https://biomejs.dev/linter/rules-sources/#rules-from-other-sources) (partially - only the "recommended" rules), which includes stuff like [no explicit-any](https://biomejs.dev/linter/rules/no-explicit-any/) and [use-for-of](https://biomejs.dev/linter/rules/use-for-of/).
> 
> If you look at the strict printed config diff in this PR you can see all the rules that are getting disabled: https://github.com/microsoft/FluidFramework/pull/22960/files#diff-7c530cc3edf251a186efaba26d97f51273af07d4fc094e7f2ea362a5b0fdb742

Hmm. I think the diffs might have been weird when #22959 was outstanding. And I think the title can be interpreted two ways.

Could be nice to add some notes to the readme about how to use the new config. I am guessing that Biome needs to be used in conjunction.

**Review by jason-ha — APPROVED** [review]

File: common/build/eslint-config-fluid/strict-biome.js
Comment (Josmithr) [inline common/build/eslint-config-fluid/strict-biome.js:7]: Can you add a section to the package README about this? It currently enumerates the other configs and explains when they should be used.

File: common/build/eslint-config-fluid/package.json
Comment (tylerbutler) [reply common/build/eslint-config-fluid/package.json:25]: Correct. In fact, no changes to any existing configs. This is a net-new config.

**Pull request merged**
